File: device-app/android/Java/umsDemo/app/src/main/java/com/rationalowl/umsdemo/data/datasource/ServerLocalDataSource.java
package com.rationalowl.umsdemo.data.datasource;

import com.rationalowl.umsdemo.data.DataDef;

public class ServerLocalDataSource extends JsonLocalDataSource<DataDef.Server> {
    private static final String FILE_NAME = "server.json";

    private static ServerLocalDataSource instance;

    public static synchronized ServerLocalDataSource getInstance() {
        if (instance == null) {
            instance = new ServerLocalDataSource();
        }

        return instance;
    }

    private ServerLocalDataSource() {
        super(DataDef.Server.class, FILE_NAME);
    }

    public DataDef.Server getServer() {
        return getValue();
    }

    public void setServer(DataDef.Server server) {
        setValue(server);
    }
}

File: device-app/ios-swift/umsDemo/ums/Data/DataSource/ServerLocalDataSource.swift
class ServerLocalDataSource: JsonLocalDataSource<DataDef.Server> {
    private static let fileName = "server.json"

    static let shared = ServerLocalDataSource()

    private init() {
        super.init(Self.fileName)
    }

    var server: DataDef.Server? {
        get {
            return value
        }
        set {
            value = newValue!
        }
    }
}
